sample_code/clean_unknown_vertices/ExprFunctions.hpp

```diff
@@ -541,6 +541,93 @@ inline string CombineTwoNumbers(int64_t one, int64_t two) {
     }
     return outputList;
   }
+
+  inline void RecoverOneEdge(
+      gpr::Context& context,
+      gshared_ptr<topology4::GraphUpdates> graphupdate,
+      uint32_t eid,
+      topology4::EdgeAttribute* eAttr,
+      int64_t src,
+      int64_t tgt) {
+    auto attr_up = graphupdate->GetEdgeAttributeUpdate(eid);
+    if (attr_up->Set(eAttr)) {
+      graphupdate->UpsertEdge(
+          topology4::DeltaVertexId(context.GraphAPI()->GetVertexType(src), src),
+          topology4::DeltaVertexId(context.GraphAPI()->GetVertexType(tgt), tgt),
+          attr_up,
+          eid);
+    } else {
+      GEngineInfo(InfoLvl::Brief, "RecoverOneEdge")
+      << "[ERROR]: Insert edge (src: "
+      << src
+      << ", tgt: "
+      << tgt
+      << ", eid: "
+      << eid
+      << ") failed.";
+    }
+  }
+
+inline void RecoverEdges(ServiceAPI* serviceapi,
+      EngineServiceRequest& request,
+      gpr::Context& context,
+      const std::string& edgeType,
+      const std::string& rEdgeType,
+      VERTEX src,
+      ListAccum<int64_t>& fEdgeList,
+      ListAccum<int64_t>& rEdgeList,
+      int64_t& missingForward,
+      int64_t& missingReverse,
+      bool dryrun) {
+
+    auto fEid = serviceapi->GetTopologyMeta()
+      ->GetEdgeTypeId(edgeType, request.graph_id_);
+    auto rEid = serviceapi->GetTopologyMeta()
+      ->GetEdgeTypeId(rEdgeType, request.graph_id_);
+
+    auto& fTgtList = fEdgeList.data_;
+    auto& rTgtList = rEdgeList.data_;
+    std::sort(rTgtList.begin(), rTgtList.end());
```

Comment:
why fTgtList doesn't need sort

Reply:
fTgtList is automatically sorted in EdgeMap. Since we deal with segments one by one and for all the edges of the same type, it is sorted by the target id